### ISSUE TITLE
Basic Authentication crashes Firefox 52

### ISSUE THREAD

**Issue by Whoops** [pfn/passifox#612]
Running Arch Linux, Firefox 52.0.1 64-bit, PassIFox 1.2.2, visiting any page that uses basic authentication immediately crashes Firefox before the dialog is shown. Everything works fine if PassIFox is disabled.

**Comment by tlaerm** [comment]
Seconded. Happens to me too. Firefox 52.0.1 (64-bit) Linux, PassIFox 1.2.2

**Comment by Whoops** [comment]
FYI, I've been using the WebExtension version provided in #595 with no issues since filing this bug.

**Comment by mendel129** [comment]
same here, windows10 - firefox 52.0.2 64b - passifox 1.2.2

https://crash-stats.mozilla.com/report/index/f4f19a4e-1d10-4ebc-aeb8-9247d2170329

example website: https://mendelonline.be/glype/

**Comment by wknd** [comment]
Same on ubuntu 16.04, Firefox 52.0.2, PassIFox 1.2.2

**Comment by NoNiMad** [comment]
Same on Windows 10, Firefox 52.0.2 (64 bits), PassIFox 1.2.2 (KeePassHttp 1.8.4.2 & KeePass 2.35 but not sure it's relevant here)

**Comment by ghost** [comment]
Confirm this bug with macOS 10.12.4, Firefox 52.0.2 and Passifox 1.2.2

**Comment by hockeymikey** [comment]
PassIFox alone doesn't crash it basic prompts, only when paired with certain addons.  I've found the following cause it:
- Zotero Connector (5.0.18)
- uBlock's new hybrid version (1.13.10) crashes it but (1.13.8) works just fine

Tested with WaterFox (54.0.3) and CyberFox (52.3.0)

**Comment by michauk** [comment]
My firefox suddendly crashed a few hours ago. I figured out my passifox plugin was involved.
As jockeymikey mentionned, this is the combination betwwen plugins, I disabled ublock (which just upgraded to 1.13.10) and switched to this good ol' adblockplus because I'm in a hurry

**Comment by davelit** [comment]
I had the same problem. I have tested @smorks' [chromeipass_firefox](https://github.com/smorks/passifox) fork, which seems to work fine so far. The only caveat is that you need to re-pair it with KeePassHTTP every time you restart your browser, as it is currently not able to persist its configuration, including the pairing key. See [Change from using localStorage to WebExtension storage API for saving prefs](https://github.com/smorks/passifox/issues/1).

**Comment by ghost** [comment]
I can confirm this with Firefox 55.0.3 on MacOS with PasslFox 1.2.1 and uBlock 1.14.0.  After I disable uBlock, the browser won't crash anymore.

**Comment by zilon84** [comment]
Same here with FF 55.0.3 on Windows 10 with PassIFox and uBlock origin. If one of the addosn is not activated, FF does not crash.

**Comment by davelit** [comment]
Yeah, just install smorks' fork, version 2.8.5 or later: https://github.com/smorks/passifox/releases. It contains the fix for the pairing issue I mentioned yesterday and works great.

In addition to that you will want to disable Firefox' built-in password store, in order to not get confused by constant [popups asking you to save your password](https://github.com/smorks/passifox/issues/3): Set "Firefox/Tools > Options > Security > Remember logins for sites" to "off".

**Comment by hockeymikey** [comment]
A fix would be nice.  Smorks' fork is a web extension version and is miles worse in featurability.

**Comment by michauk** [comment]
yes, it happened for me yesterday with adblock+ instead of ublock...
and that fork isn't efficient as passifox
bad times for firefox plugins.... I had to move to the latest firefox ESR version in order not to loose all my plugins with that new quantum. But still, this bug.... aaaaaaaaaaaaaah
maybe time to move back to chrome.......

**Comment by hockeymikey** [comment]
@michau Chrome is still miles worse.  I'd recommend Waterfox, Basilick, and Palemoon in that order.

**Comment by michauk** [comment]
I moved to quantum. I lost 2 hours but managed to find every plugin I needed (still tabmix+ missing, I'll do without until it's available). 
About keepass, I'm using "Keefox", it does the job. I had to adjust some things regarding URL wildcard (to handle subdomains authentication efficiently on a specific domain I have) + ublock origin.
Regarding firegestures (another on I couldn't skip) => I discovered strokeplus which kind of solve the plugin problem entirely! take a look at it.

keefos has a nice behaviour to fill forms in + apache authentication

Regards
